client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -233,99 +240,122 @@ def predict(
+            backoff_factor: Optional[float] = http_session.DEFAULT_BACKOFF_FACTOR
+            ) -> Tuple[str, Dict[str, Any]]:
+        """
+        Makes a prediction using input `x` the same as `predict`, but returns a tuple including the ID of the
+        prediction request along with the prediction results.
+
+        .. versionadded:: 0.23.0
+           The `prediction_id` parameter.
+        .. versionadded:: 0.23.0
+           The `retry_status` parameter.
+        .. versionadded:: 0.23.0
+           The `backoff_factor` parameter.
+
+        Parameters
+        ----------
+        x : list
+            A batch of inputs for the model.
+        compress : bool, default False
+            Whether to compress the request body.
+        prediction_id: str, default None
+            A custom str to use as the ID for the prediction request.  Defaults to a randomly generated numeric id.
+        max_retries : int, default 13
+            Maximum number of times to retry a request.
+        retry_status : set, default {404, 429}
+            Set of status codes, as integers, for which retry attempts should be made.  Overwrites default value of
+            {404, 429}.  Expand the set to include more: ``retry_status={404, 429, <NEW_STATUS_CODE>}``
+        backoff_factor : float, default 0.3
+            A backoff factor to apply between retry attempts.  Uses standard urllib3 sleep pattern:
+            ``{backoff factor} * (2 ** ({number of total retries} - 1))`` with a maximum sleep time between requests of
+            120 seconds.
+        Returns
+        -------
+        id, prediction: Tuple(str, list)
+            The first element of the tuple is the prediction ID.
+            The second element of the tuple is the output returned by the deployed model for `x`.
+
+        Raises
+        ------
+        RuntimeError
+            If the deployed model encounters an error while running the prediction.
+        requests.HTTPError
+            If the server encounters an error while handing the HTTP request.
+
+        """
+        # Set the retry config if it differs from current config.
+        self._session = http_session.set_retry_config(
+            self._session,
+            max_retries=max_retries,
+            status_forcelist=retry_status,
+            backoff_factor=backoff_factor,
+            )
+
+        # Remove request id header from previous requests if present.
+        if 'verta-request-id' in self._session.headers.keys():
+            http_session.update_session_headers(
+                self._session,
+                {'verta-request-id': ''},
+                remove=True
+                )
+
+        if prediction_id:
+            http_session.update_session_headers(
+                self._session,
+                {'verta-request-id': prediction_id},
                 )
+        response = self._predict(x, compress)
+        id = response.headers['verta-request-id']
```

Comment:
My instincts tell me that we should gracefully guard against potential backend misbehavior by not always assuming `"verta-request-id"` will be returned.

```suggestion
        id = response.headers.get("verta-request-id")
```

(This was alright before, because a user had to explicitly call `predict_with_id()`, but now we're checking the header on all `predict()` calls)